tests/nativeapp/test_init.py

```diff
@@ -1,60 +1,139 @@
 PROJECT_NAME = "demo_na_project"
-MAX_ALLOWED_NUM_CHARACTERS = 255
+CUSTOM_GIT_REPO_URL = "https://testing.com/my-native-app-template"
+
+
+SNOWFLAKE_YML = dedent(
+    """\
+    definition_version: 1
+    native_app:
+        name: demo_fixed_na_project
+        artifacts:
+            - app/setup_script.sql
+"""
+)
+
+TEMPLATED_SNOWFLAKE_YML = dedent(
+    """\
+    definition_version: 1
+    native_app:
+        name: {{project_name}}
+        artifacts:
+            - app/setup_script.sql
+"""
+)
+
+# --------------------------------------
+# ----- Test Harness Helpers -------
+# --------------------------------------
+
+
+def fake_clone_template_with_files(files: dict):
+    def fake_clone_mock(url: str, to_path: str, filter: list, depth: int):
+        repo_path = Path(to_path)
+        repo_path.mkdir(parents=True, exist_ok=True)
+
+        # create a fake .git directory
+        git_dir_path = repo_path / ".git"
+        git_dir_path.mkdir()
+
+        for file_name in files:
+            file_contents = files[file_name]
+            create_named_file(
+                file_name=file_name,
+                dir=str(repo_path),
+                contents=[file_contents],
+            )
+
+    return fake_clone_mock
+
+
+def fake_clone_template_with_file(file_name: str, file_contents: str):
+    return fake_clone_template_with_files({file_name: file_contents})
+
+
+def fake_clone_default_repo(url: str, to_path: str, filter: list, depth: int):
+    assert url == "https://github.com/snowflakedb/native-apps-templates"
+    repo_path = Path(to_path)
+    repo_path.mkdir(parents=True, exist_ok=True)
+
+    # create a fake .git directory
+    git_dir_path = repo_path / ".git"
+    git_dir_path.mkdir()
+
+    # create a fake basic template
+    basic_template_dir = repo_path / "basic"
+    basic_template_dir.mkdir()
+    create_named_file(
+        file_name="snowflake.yml.jinja",
+        dir=str(basic_template_dir),
+        contents=[TEMPLATED_SNOWFLAKE_YML],
+    )
+
+    # create a fake python-streamlit template
+    py_template_dir = repo_path / "python-streamlit"
+    py_template_dir.mkdir()
+    create_named_file(
+        file_name="snowflake.yml.jinja",
+        dir=str(py_template_dir),
+        contents=[TEMPLATED_SNOWFLAKE_YML],
+    )
+
+
+def fake_clone_jinja_template_repo(url: str, to_path: str, filter: list, depth: int):
+    fn = fake_clone_template_with_file("snowflake.yml.jinja", TEMPLATED_SNOWFLAKE_YML)
+    return fn(url=url, to_path=to_path, filter=filter, depth=depth)
+
+
+def fake_clone_template_repo(url: str, to_path: str, filter: list, depth: int):
+    fn = fake_clone_template_with_file("snowflake.yml", SNOWFLAKE_YML)
+    return fn(url=url, to_path=to_path, filter=filter, depth=depth)
+
 
 # --------------------------------------
 # ----- Tests for Helper Methods -------
 # --------------------------------------
 
 
 @pytest.mark.parametrize(
-    "project_name, expected",
+    "s,expected",
```

Comment:
`s` is not a good variable name :)

Reply:
renamed s,expected -> python_string,yaml_string

How does that sound?